Fix data split delimiter

ota42y opened the pull request:
IRC Protocol use CR-LF, but legacy IRC server send other code.
http://tools.ietf.org/html/rfc1459.html#section-8

So, replace split string from \r\n with /\r\n|\r|\n/

jirwin commented:
+1

apeiron commented:
I'm a little wary of the precedence of the | in this regex and I want to torture-test it a bit before merging.

apeiron commented:
I do not think this regex works in full generality. Consider the following:

```
var string = 'hello\nthere\nworld\r\nstuff'
var badRe = new RegExp('/\r\n|\r|\n/')
console.dir(string.split(badRe));
var goodRe = new RegExp('(?:\r\n)|\n|\r')
console.dir(string.split(goodRe));
```
Can you consider the second regex in this example for your pull req, or point out where I've misunderstood? Thanks!

jirwin commented:
@apeiron Great catch. We may also want to consider compiling the regex out of scope of the listener so it doesn't have to be created for every chunk of data coming in.

ota42y commented:
@jirwin Ok, I moved the regex.

@apeiron  thanks! I overlooked that's pattern.

But, I think you should delete `/` word from the badRe, and it's worked well.

```javascript
var string = 'hello\nthere\nworld\r\nstuff'
var badRe = new RegExp('\r\n|\r|\n')
var badResult = string.split(badRe)
console.dir(badResult);
var goodRe = new RegExp('(?:\r\n)|\n|\r')
var goodResult = string.split(goodRe);
console.dir(goodResult)

var isEqual = (badResult.length == goodResult.length)

for (var i = 0; i < badResult.length; ++i) {
 if (badResult[i] !== goodResult[i]){
   isEqual = false
 }
}

console.dir(isEqual) # true
```

I think `/\r\n|\r|\n/` isn't equal to`\r\n|\r|\n` in RegExp.

Probably, the RegExp object interpret it as `\/\r\n|\r|\n\/`.
Because, the result executing `string.split(/\/\r\n|\r|\n\//)` is equal to executing `string.split(new RegExp('/\r\n|\r|\n/'))`.

So I fixed code using `new RegExp('\r\n|\r|\n')`

apeiron commented:
This needs rebasing, because we just merged #278.

ota42y commented:
I rebased.

jirwin commented:
+1
